### ISSUE TITLE
Missing `GetDouble` in C# SDK

### ISSUE THREAD

**Issue by iwahbe** [pulumi/pulumi#16343]
### What happened?

I was trying to generate a C# SDK for a dynamic provider and it failed to compile.

### Example

```console
𝛌 pulumi package gen-sdk --language csharp  bin/pulumi-resource-terraform-bridge CiscoDevNet/ise
𝛌 (cd sdk/dotnet && dotnet build)                                                           
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  Restored /Users/ianwahbe/go/src/github.com/pulumi/pulumi-terraform-bridge/dynamic/sdk/dotnet/Pulumi.Ise.csproj (in 100 ms).
/Users/ianwahbe/go/src/github.com/pulumi/pulumi-terraform-bridge/dynamic/sdk/dotnet/Config/Config.cs(55,97): error CS1061: 'Config' does not contain a definition for 'GetDouble' and no accessible extension method 'GetDouble' accepting a first argument of type 'Config' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi-terraform-bridge/dynamic/sdk/dotnet/Pulumi.Ise.csproj]

Build FAILED.

/Users/ianwahbe/go/src/github.com/pulumi/pulumi-terraform-bridge/dynamic/sdk/dotnet/Config/Config.cs(55,97): error CS1061: 'Config' does not contain a definition for 'GetDouble' and no accessible extension method 'GetDouble' accepting a first argument of type 'Config' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi-terraform-bridge/dynamic/sdk/dotnet/Pulumi.Ise.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.88
```

### Output of `pulumi about`

```
CLI          
Version      3.119.1-dev.0
Go Version   go1.22.3
Go Compiler  gc

Host     
OS       darwin
Version  14.5
Arch     arm64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/ian-pulumi-corp
User           ian-pulumi-corp
Organizations  ian-pulumi-corp, pulumi
Token type     personal

Pulumi locates its logs in /var/folders/fg/_1q36r4j6yx0rwz2fbhjd5y40000gn/T/ by default
```

### Additional context

I talked to Fraser about this earlier and he suspects this is because we are allowing a
any version `>= Pulumi 3.0.0`.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by iwahbe** [comment]
The full schema generated can be found at https://gist.github.com/iwahbe/b6aa308f0ecdbdf9993f6babb3b4b402.

**Comment by justinvp** [comment]
Other providers we ship have the following in the schema (e.g. [random](https://github.com/pulumi/pulumi-random/blob/ecd3b79bb8642b4aad5ec80b5b88be14f6cf8f26/provider/cmd/pulumi-resource-random/schema.json#L16-L19)):

```json
            "packageReferences": {
                "Pulumi": "3.*"
            },
```

This schema doesn't have that. Which should be fine. We should fill in a default value, in that case:

https://github.com/pulumi/pulumi/blob/78c48204e056d58217bda37c187e2ade5929c040/pkg/codegen/dotnet/gen.go#L2260-L2264

I haven't tried it out. Are we not filling that default value in? Is the version range incorrect?

@Zaid-Ajaj, could you take a look?

**Comment by justinvp** [comment]
The problem is the default version range of `[3.54.1.0,4)`. `GetDouble` was added and first shipped in 3.55 (https://github.com/pulumi/pulumi-dotnet/commit/380536dd851d74304969dbc3c7d7180d1c59fe06).

**Comment by cleverguy25** [comment]
Added to epic https://github.com/pulumi/home/issues/3371
